vars/AbstractMDSDToolsPipeline.groovy

```diff
@@ -0,0 +1,106 @@
+/**
+ * Abstract MDSDTools Pipeline
+ * Defines pipeline stages of MDSD.tools Build process
+ * Relies on concrete Pipeline to provide agent configuration
+ */
+def call(defaults  = [:], overrides = [:]) {
+  extendConfiguration(defaults, [    
+    deployRelease: params.Release,
+    deployReleaseVersion: params.ReleaseVersion
+  ] + overrides) 
+
+  def config = updateConfiguration()
+  
+  def modules = ['Prepare', 'Checkout', 'Build', 'Archive', 'Quality Metrics', 'Deploy', 'Cleanup']
+  def moduleConfig = [:]
+
+  modules.each {
+    if (!config["skip${it.replaceAll("\\s","")}"]) {
+      moduleConfig.put(it, [:])
+    } else {
+      echo "Stage $it will be skipped."
+    }
+  }
+  
+  extendConfiguration([
+    modules: moduleConfig,
+    notifyDefault: decodeEmailAddress(config.notifyDefault)
```

Comment:
module config, as we calculate it from the skipPhase keys
notifyDefault, as it could be provided by either the Jenkinsfile (through defaults) or through overrides which is fused in the first extendConfiguration step. Here the parameter is converted only if the email address is base64 encoded.